File: joyal670-protenium-iro-hub-9d65cb9ae2c4/app/src/main/java/com/protenium/irohub/ui/main/dashboard/home/adapter/OffDaysAdapter.java
package com.protenium.irohub.ui.main.dashboard.home.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.protenium.irohub.R;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class OffDaysAdapter extends RecyclerView.Adapter<OffDaysAdapter.ViewHold> {

    private List<String> stringList;
    private Context context;
    private Boolean clicked;

    public OffDaysAdapter(Context fragment, List<String> dataList) {
        this.stringList = dataList;
        this.context = fragment;
    }

    @NonNull
    @Override
    public ViewHold onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.recycerview_offdays, parent, false);
        return new ViewHold(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHold holder, int position) {

        holder.offDaysBtn.setText(stringList.get(position));

    }

    @Override
    public int getItemCount() {
        return stringList.size();
    }

    public class ViewHold extends RecyclerView.ViewHolder {

        @BindView(R.id.offDaysBtn)
        TextView offDaysBtn;


        public ViewHold(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }

}

File: proteinium/app/src/main/java/com/proteinium/proteiniumdietapp/ui/main/home/home/detailed_page/adapter/OffDaysAdapter.kt
package com.proteinium.proteiniumdietapp.ui.main.home.home.detailed_page.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.proteinium.proteiniumdietapp.R
import com.proteinium.proteiniumdietapp.utils.CommonUtils
import kotlinx.android.synthetic.main.recycerview_offdays.view.*


class OffDaysAdapter(var offDays: List<String>) : RecyclerView.Adapter<OffDaysAdapter.ViewHold>() {

    private var context: Context? = null


    class ViewHold(itemView: View) : RecyclerView.ViewHolder(itemView)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHold {
        context = parent.context
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.recycerview_offdays, parent, false)
        return ViewHold(
            view
        )
    }

    override fun getItemCount(): Int {
        return offDays.size
    }

    override fun onBindViewHolder(holder: ViewHold, index: Int) {
        val  data=StringBuilder()
        data.append( CommonUtils.NonStopDays2(offDays[index], context!!) )
        if(index!=offDays.size-1){
            data.append(",")
        }
        holder.itemView.offDaysBtn.text =data.toString()

    }
}